Adding functionality to set frame index directly

liuloppan opened the pull request:
- Adding functionality to set the intital "current frame index" by passing it as a prop into Carousel.
- Adding the function setFrame to Carousel and passing it as a Widget prop ('setFrameHandler') which can be used by Indicator dots to transition multiple frames at once.

liuloppan commented:
#125 
#100

amio commented:
Thanks @liuloppan! will look into it soon :D

amio commented on src/carousel.js:
```diff
@@ -262,21 +263,37 @@ class Carousel extends React.Component {
+    const diff = Math.abs(index - this.state.currentFrameIndex);
+    if (index < this.state.currentFrameIndex) {
+      for (let i = 0; i < diff; i++) {
+        setTimeout(() => this.prev(), i * ms);
```
There would be a series of animation for switching to target frame, which works but not ideal, especially when user jumps between a lot frames. We need to switch to target frame directly. 🤔

liuloppan replied:
Ok I preferred to have the animation and to be able to control the animation time. For a small number of frames setting 'ms' to 0 will visually have the same effect, but for many frames this could be an issue as you say. But I will look into creating a function that allows changing it directly.

liuloppan replied:
@amio I have updated the PR. Also updated index.js so that the example is using this functionality. Currently setFrameIndex is set to 4 to show how it works, but it could be removed.

liuloppan commented:
> @liuloppan Great! Just a few changes:
> 
> * since `loop` cannot work with `currentFrameIndex`, we should log a warning in case of that,
> * let's keep the default demo as original, add some description of `currentFrameIndex` in README.md

* `loop` works with my intended use of `currentFrameIndex`, I renamed it to  `onStartFrameIndex` and added description in README.md 
* I changed back to the original demo index.js
